src/snackbar/snackbar.js

```diff
@@ -0,0 +1,190 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+(function() {
+  'use strict';
+
+  /**
+   * Class constructor for Snackbar MDL component.
+   * Implements MDL component design pattern defined at:
+   * https://github.com/jasonmayes/mdl-component-design-pattern
+   *
+   * @constructor
+   * @param {HTMLElement} element The element that will be upgraded.
+   */
+  var MaterialSnackbar = function MaterialSnackbar(element) {
+    this.element_ = element;
+    this.active = false;
+    this.init();
+  };
+  window['MaterialSnackbar'] = MaterialSnackbar;
+
+  /**
+   * Store strings for class names defined by this component that are used in
+   * JavaScript. This allows us to simply change it in one place should we
+   * decide to modify at a later date.
+   *
+   * @enum {string}
+   * @private
+   */
+  MaterialSnackbar.prototype.cssClasses_ = {
+    snackbar: 'mdl-snackbar',
+    message: 'mdl-snackbar__text',
+    action: 'mdl-snackbar__action',
+    activeSnackbar: 'is-active'
+  };
+
+  /**
+   * Create the internal snackbar markup.
+   *
+   * @private
+   */
+  MaterialSnackbar.prototype.createSnackbar_ = function() {
+    this.snackbarElement_ = document.createElement('div');
+    this.textElement_ = document.createElement('div');
+    this.snackbarElement_.classList.add(this.cssClasses_.snackbar);
+    this.textElement_.classList.add(this.cssClasses_.message);
+    this.snackbarElement_.appendChild(this.textElement_);
+    this.snackbarElement_.setAttribute('aria-hidden', true);
+
+    if (this.actionHandler_) {
+      this.actionElement_ = document.createElement('button');
+      this.actionElement_.type = 'button';
+      this.actionElement_.classList.add(this.cssClasses_.action);
+      this.actionElement_.textContent = this.actionText_;
+      this.snackbarElement_.appendChild(this.actionElement_);
+      this.actionElement_.addEventListener('click', this.actionHandler_);
+    }
+
+    this.element_.appendChild(this.snackbarElement_);
+    this.textElement_.textContent = this.message_;
+    this.snackbarElement_.classList.add(this.cssClasses_.activeSnackbar);
+    this.snackbarElement_.setAttribute('aria-hidden', false);
+    setTimeout(this.cleanup_.bind(this), this.timeout_);
+
+  };
+
+  /**
+   * Remove the internal snackbar markup.
+   *
+   * @private
+   */
+  MaterialSnackbar.prototype.removeSnackbar_ = function() {
+    if (this.actionElement_ && this.actionElement_.parentNode) {
+      this.actionElement_.parentNode.removeChild(this.actionElement_);
+    }
+    this.textElement_.parentNode.removeChild(this.textElement_);
+    this.snackbarElement_.parentNode.removeChild(this.snackbarElement_);
+  };
+
+  /**
+   * Create the internal snackbar markup.
+   *
+   * @param {Object} data The data for the notification.
```

Comment:
There's two options here: either we add stricter typing for the data object, or we keep things as they are but switch to a different syntax below. How open-ended do you see this data object, i.e., would you prefer type-safety or flexibility?

If you prefer type-safety, you can type this a bit more strictly as done in e.g. https://github.com/google/material-design-lite/blob/master/src/mdlComponentHandler.js#L422 and https://github.com/google/material-design-lite/blob/master/src/mdlComponentHandler.js#L268

If you prefer flexibility (and simpler code), read below :)